slash/parallel/worker.py

```diff
@@ -58,34 +60,36 @@ def start_execution(self, app, collected_tests):
             os.setsid()
         register(self.warning_added)
         register(self.error_added)
-        collection = [(test.__slash__.file_path, test.__slash__.function_name, test.__slash__.variation.id) for test in collected_tests]
-
-        if not self.client.validate_collection(self.client_id, collection):
+        collection = [(test.__slash__.file_path, test.__slash__.function_name, json.dumps(test.__slash__.variation.id)) for test in collected_tests]
+        if not self.client.validate_collection(self.client_id, sorted(collection)):
```

Comment:
There's still a problem here... the sorting order actually depends on the construction order of the variation dict, which means we still have a problem. I think the sorting part should use the ordered dictionary item tuples...